### PR TITLE
Add analyze open documents only

### PR DESCRIPTION
**Pull request by neoGeneva**
Hey, this is a pull requestion to add an `AnalyzeOpenDocumentsOnly` option to `RoslynExtensionsOptions`, which `CSharpDiagnosticWorkerWithAnalyzer` uses to decided if it runs full roslyn anaylizers or just the base diagnostics provided by  `syntaxTree.GetDiagnostics()`.

The main benefit is the speed of the initial analysis of all files in the solution. For large solutions this is pretty handy, for example the roslyn solution takes 18 minutes for that first pass, vs. 1 minute with `AnalyzeOpenDocumentsOnly` set to true.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2346) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2346) before we can accept your contribution.<br/><br/>:x: neoGeneva [sign now](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2346)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OmniSharp/omnisharp-roslyn?pullRequest=2346) it.</sub>

**Review by filipw — APPROVED** [review]
LGTM, thanks! - any thoughts @333fred @JoeRobich ?

File: tests/OmniSharp.Roslyn.CSharp.Tests/CustomRoslynAnalyzerFacts.cs [hunk 1]
```diff
@@ -258,7 +258,42 @@ public async Task WhenDiagnosticsRulesAreUpdated_ThenReAnalyzerFilesInProject()
...
         }
 
-        private ProjectId AddProjectWitFile(OmniSharpTestHost host, TestFile testFile, TestAnalyzerReference testAnalyzerRef = null)
+
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public async Task WhenDocumentIsntOpenAndAnalyzeOpenDocumentsOnlyIsSet_DontAnylizeFiles(bool analyzeOpenDocumentsOnly, bool isDocumentOpen)
```
Comment (JoeRobich) [inline tests/OmniSharp.Roslyn.CSharp.Tests/CustomRoslynAnalyzerFacts.cs:267]:
```suggestion
        public async Task WhenDocumentIsntOpenAndAnalyzeOpenDocumentsOnlyIsSet_DontAnalyzeFiles(bool analyzeOpenDocumentsOnly, bool isDocumentOpen)
```

**Review by JoeRobich — APPROVED** [review]

**Review by 333fred — APPROVED** [review]

**Comment by neoGeneva** [comment]
Wooo! Thanks so much for this folks, this is my first real contribution to a public repo, feels good to help out ❤️

**Comment by 333fred** [comment]
@neoGeneva, would you want to submit a PR to the vscode to add the option? You'll need to:

* Add the setting to the dictionary, like this: https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/omnisharp/server.ts#L390-L392
* Add the setting to the options type, with a default: https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/omnisharp/options.ts#L8
* Add the setting to the package.json with a title and a description, like this: https://github.com/OmniSharp/omnisharp-vscode/blob/master/package.json#L993-L997
* Adjust this test: https://github.com/OmniSharp/omnisharp-vscode/blob/master/test/unitTests/options.test.ts#L13-L39
* Adjust this constructor: https://github.com/OmniSharp/omnisharp-vscode/blob/master/test/unitTests/Fakes/FakeOptions.ts#L9

My suggestion would be off (false) by default for this setting.

**Comment by neoGeneva** [comment]
Sweet, thanks @333fred, I've got most of it ready to go, but it got mixed up with some other stuff I was looking at, just splitting it out into its own branch so I can do a pull request a little later.

**Pull request merged**